packages/@aws-cdk/aws-codepipeline/lib/stage.ts

```diff
@@ -126,7 +117,7 @@ export class Stage extends cdk.Construct implements actions.IStage, actions.IInt
   }
 
   public onStateChange(name: string, target?: events.IEventRuleTarget, options?: events.EventRuleProps) {
-    const rule = new events.EventRule(this.pipeline, name, options);
+    const rule = new events.EventRule((this.pipeline as Pipeline), name, options);
```

Comment:
`this.pipeline as Pipeline` should just be `this`

Reply:
No, it should not, as `events.EventRule` takes a `cdk.Construct` as its first parameter, while `stage.pipeline` is of type `IPipeline` (we cannot make it of type `Pipeline`, which would make this cast unnecessary, as JSII does not allow implementing an interface property with a subtype of the interface declared type).

Reply:
Oh, did you mean to change the parent of the event to the Stage instead of Pipeline?

Reply:
Yap. Always pass “this”. It’s a simple rule of thumb :-)

Reply:
Done in the latest revision.